src/libsyntax/parse/parser.rs

```diff
@@ -5970,13 +5970,12 @@ impl<'a> Parser<'a> {
                                  &token::CloseDelim(token::Brace),
                                  SeqSep::trailing_allowed(token::Comma), |this| {
             let lo = this.span.lo;
-            let node = if this.eat_keyword(keywords::SelfValue) {
-                let rename = this.parse_rename()?;
-                ast::PathListItemKind::Mod { id: ast::DUMMY_NODE_ID, rename: rename }
-            } else {
-                let ident = this.parse_ident()?;
-                let rename = this.parse_rename()?;
-                ast::PathListItemKind::Ident { name: ident, rename: rename, id: ast::DUMMY_NODE_ID }
+            let ident = this.parse_path_segment_ident()?;
```

Comment:
Having the parser accept `super` and `Self` in import lists mildly regresses diagnostics (imo), so I amended the refactoring so that the parser continues to not accept `super` and `Self`.